src/scripts/main.js

```diff
@@ -1,5 +1,33 @@
 'use strict';
 
 document.addEventListener('click', e => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+
+  const maxWidth = wall.clientWidth - spider.clientWidth;
+  const maxHeight = wall.clientHeight - spider.clientHeight;
+
+  const x = e.offsetX - spider.clientWidth / 2;
+  const y = e.offsetY - spider.clientHeight / 2;
+
+  if (!e.target.closest('.wall')) {
+    return;
+  };
+
+  function wallLimit(value, limit) {
+    switch (true) {
+      case (value < 0):
+        return 0;
+      case (value > limit):
+        const current = limit;
+
+        return current;
+
+      default:
+        return value;
+    };
+  };
```

Comment:
You can do it simpler:
```
function handleValue(value, limit) {
  if (value < 0) {
    return 0;
  }

  if (value > limit) {
    return limit;
  }

  return value;
}
```

It's more readable for me

Reply:
ok